include/gunrock/framework/operators/filter/predicated.hxx

```diff
@@ -82,46 +83,43 @@ void execute_mgpu(graph_t& G,
   input_ends[num_gpus - 1] = input->end();
   
   // Map
-  size_t new_sizes[num_gpus];
+  int new_sizes[num_gpus];
   for(int i = 0; i < num_gpus; i++) {
     auto ctx = context.get_context(i);
     cudaSetDevice(ctx->ordinal());
     
-    auto new_length = thrust::copy_if(
+    auto new_end = thrust::copy_if(
       thrust::cuda::par.on(ctx->stream()),
       input_begins[i],
       input_ends[i],
       output_begins[i],
       predicate
     );
     
-    new_sizes[i] = thrust::distance(output_begins[i], new_length);
+    new_sizes[i] = (int)thrust::distance(output_begins[i], new_end);
     cudaEventRecord(ctx->event(), ctx->stream());
   }
-  
+
   // Sync
   auto context0 = context.get_context(0);
   for(int i = 0; i < num_gpus; i++) {
     auto ctx = context.get_context(i);
     cudaStreamWaitEvent(context0->stream(), ctx->event(), 0);
```

Comment:
This is actually syncing everything to stream/process of the first GPU (`GPU 0`), I don't think we want that. In the Gunrock ForAll code that Serban wrote, the user-provided stream is like a master-user stream that everything synchronizes to because we do not want new operators to run on the user stream without getting all the result from a multi-gpu forall.

In essentials case however, we can leave them completely asynchronous until we need to `reduce` in the following step.